### ISSUE TITLE
Support for questions.yaml ?

### ISSUE THREAD

**Issue by michelpromonet** [komodorio/helm-dashboard#464]
Hi,

I was wondering if it is possible to use questions.yaml like in rancher ui https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/helm-charts-in-rancher/create-apps#additional-files-for-rancher-charts, or have a similar feature ?

Best Regards,
Michel.

**Comment by undera** [comment]
I don't want to say hard "no" to this, here are my thoughts:
- Rancher chart style will have smaller addressable audience, thus it raises question of "is it efficient to invest into this specific case, or better to invest into general `values.yaml` editing flow"
- Many other projects focus on presenting values.yaml editing via generated forms, none made it perfect. IMO it becomes a mess when there are many values to find and specify.
- I claim that current install/upgrade dialog is already problematic UI-wise, adding any more complication to it will make it worse
- Initially the design approach I was taking is to be lean and not create a complicated (and more glitchy) UI
- There is old idea from times when HD was just born, that if we would edit values.yml like text with comments, then it would be simple, but efficient

To summarize, suggestion like this could have place, if the install/edit dialog would allow for switching between different "modes" of editing properties. Right now there's no place to apply it in the UI. We'd need some design ideas of how to rearrange the dialog with that in mind. This is a good way to contribute to project: just bring UI mocks of possible improvements.

**Comment by michelpromonet** [comment]
Hi @undera,

Thanks for your answer, I understand your point of view. I was mentionned rancher as a sample of an easy way to configure some of variables of the values.yaml. 

Best Regards,
Michel.

**Comment by garethsb** [comment]
Hi @undera,

I'm interested in something which seems related, whether helm-dashboard could take advantage of the optional [Helm v3 _values.schema.json_](https://helm.sh/docs/topics/charts/#schema-files), when provided by a Helm chart, for two things:
* Validating user-supplied _values.yaml_ - see e.g. https://www.arthurkoziel.com/validate-helm-chart-values-with-json-schemas/
* Providing form-based view/mode for editing _values.yaml_ - see e.g. https://gimlet.io/blog/helm-react-ui-a-react-component-to-render-ui-for-helm-charts

Is this something that is already on the roadmap for helm-dashboard or that you would accept a PR for?

Thanks

**Comment by undera** [comment]
@garethsb the validation part should happen as-is, because the backend runs dry-run process when previewing diff

For the rich UI building - I'm not big fan of it. IMO it takes a lot of visual space on screen and cumbersome to do right. While it is really needed for new Helm users, which eventually get experienced and fine to use YAML.

That said, it is Open Source, we can look at proposed implementation and discuss it.
